### DISCUSSION TITLE
use `:where` in reset-stylesheet

### DISCUSSION THREAD

**Discussion by inoas** [tailwindlabs/tailwindcss#8444] [Ideas]
This will allow overriding with `:where` later on and keep the specificity low.

Right now tailwind is loaded then some custom file that sets some defaults for h1-h6 and I cannot use `:where()` there anymore because tailwind already sets that.

https://caniuse.com/mdn-css_selectors_where
https://developer.mozilla.org/en-US/docs/Web/CSS/:where

Another way to do this would be separating the reset parts from the rest of tailwind so that one can opt out of those easily

**Comment by inoas** [comment]
I see `tailwindcss/base` can be opted out already.

I will take a a look how it goes and if there is a good way to reset without adding specificity in tailwind already.

**Comment by inoas** [comment]
https://developer.mozilla.org/en-US/docs/Web/CSS/Specificity#the_where_exception

**Reply by clord** [reply]
I also would love to have selectors wrapped in `:where` in `@tailwindcss/forms` (or really all of the official plugins). It would mean the reset styles like `input[type="email"]` will not win over my own global `:where` resets. 

Ideally I'd define some base `@apply` rules for built-in nodes like `input`, `button`, etc., and then override with `class="p-5"` for customization. the class selector loses to attribute selectors though, unless attribute selector is in `:where`. Tried that, but then built in plugin rules win!

`:where` really lets reset styles shine, since they will never win like this, but still win against browser default agent rules.

**Reply by inoas** [reply]
At least it would be an awesome flag to be set globally per project so that everything is either wrapped in where or not, thus we can opt into it given we are only at around 90% support https://caniuse.com/?search=%3Awhere

**Reply by inoas** [reply]
... and later some breaking tailwind release could switch from default otp-in of where to default opt-out of where and later yet remove that flag if it is deemed unnecessary then.

**Reply by clord** [reply]
I'm just learning of `@layer`, as well, not sure which is better way to solve this... perhaps within a layer, a reset should still be `:where`?

**Reply by inoas** [reply]
`@layer` is by 85% so even less than `:where`'s 90%:
https://caniuse.com/?search=layer
